### ISSUE TITLE
Fix: Switching from snowflake to fully random generated socket id

### ISSUE THREAD

**Issue by Totodore** [Totodore/socketioxide#2]
Currently the socket id in the engine.io crate is generated with a snowflake ID. 

However it means that socket ids are predictable whereas for security reasons they should be fully random (like in the socket.io server with a base64 20 char ID).

base64 Ids should have a fixed size and represented as an array of bytes to avoid heap allocation (like String).

A type alias would be good to unify SID type around the codebase.

**Comment by biryukovmaxim** [comment]
Hey, I think that's not flexible enough, go and js implementations use String, but I can suggest using generic Clone + Eq + Hash. If type is Clone it is easy to implement Clone for it. 
It cab be implemented with Generator trait
```
pub trait Generator: Clone {
    type Sid: Clone + Hash + Eq + Debug;
    fn generate_sid(&self) -> Self::Sid;
}
```
What do you think about it?

**Comment by Totodore** [comment]
I like the idea of a `Generator` trait so it allows the user to specify a custom generator function.

But it would imply to propagate another  generic all around the codebase.

Idk if it is worth it but why not try and see  the result !

**Comment by biryukovmaxim** [comment]
I created a PR.
I haven't fixed docs test, it expects addition of type annotations.
From my point  of view it can make project more complicated and less developer friendly.
Could you check it out? Now I'm not sure if it's really worth it

**Comment by Totodore** [comment]
As I said in your PR, it is maybe not the best moment to add so much complexity to the codebase due to the generic.
However you can still make a fixed implementation of a fully random, fixed size generated socket id. It is the part that's bother me the most because snowflakes are predictible.

**Comment by Totodore** [comment]
Resolved with #19
